### PR TITLE
Updating hemv, her, her2, hpmv, hpr, hpr2 tests

### PR DESCRIPTION
**Pull request by daineAMD**
This PR adds hemv, her, her2, hpmv, hpr, and hpr2 to hipblas bench. It also adds device_mode tests, updates to use the new handle, and updates to use more recent batched arrays, along with any other cleanup I noticed.

### CODE REVIEW

File: clients/include/testing_hpr.hpp [hunk 1]
```diff
@@ -37,71 +36,94 @@ hipblasStatus_t testing_hpr(const Arguments& argus)
...
+        for(int iter = 0; iter < runs; iter++)
+        {
+            if(iter == argus.cold_iters)
+                gpu_time_used = get_time_us_sync(stream);
+
+            CHECK_HIPBLAS_ERROR(hipblasHprFn(handle, uplo, N, d_alpha, dx, incx, dA));
         }
+        gpu_time_used = get_time_us_sync(stream) - gpu_time_used;
+
+        ArgumentModel<e_N, e_incx>{}.log_args<T>(std::cout,
```
Comment (TorreZuk) [inline clients/include/testing_hpr.hpp:119]: Might want to add e_alpha, as we log it in rocBLAS?  If alpha is 0 it explains the high GFLOPs.

File: clients/include/testing_hpr2.hpp [hunk 2]
```diff
@@ -38,75 +36,100 @@ hipblasStatus_t testing_hpr2(const Arguments& argus)
...
+        for(int iter = 0; iter < runs; iter++)
+        {
+            if(iter == argus.cold_iters)
+                gpu_time_used = get_time_us_sync(stream);
+
+            CHECK_HIPBLAS_ERROR(hipblasHpr2Fn(handle, uplo, N, d_alpha, dx, incx, dy, incy, dA));
         }
+        gpu_time_used = get_time_us_sync(stream) - gpu_time_used;
+
+        ArgumentModel<e_N, e_incx, e_incy>{}.log_args<T>(std::cout,
```
Comment (TorreZuk) [inline clients/include/testing_hpr2.hpp:125]: e_alpha ?

File: clients/include/testing_hpr_strided_batched.hpp [hunk 3]
```diff
@@ -48,74 +47,106 @@ hipblasStatus_t testing_hpr_strided_batched(const Arguments& argus)
...
+        {
+            if(iter == argus.cold_iters)
+                gpu_time_used = get_time_us_sync(stream);
+
+            CHECK_HIPBLAS_ERROR(hipblasHprStridedBatchedFn(
+                handle, uplo, N, d_alpha, dx, incx, stride_x, dA, stride_A, batch_count));
         }
+        gpu_time_used = get_time_us_sync(stream) - gpu_time_used;
+
+        ArgumentModel<e_N, e_incx, e_stride_x, e_stride_a, e_batch_count>{}.log_args<T>(
```
Comment (TorreZuk) [inline clients/include/testing_hpr_strided_batched.hpp:141]: Double check if the ` log_args<T>`  should be U for alpha logging (which could be logged) as the GLOPs and GBYTES already get T separately.

**Review by TorreZuk — APPROVED** [review]
Looks good, minor questions raised.

**Comment by daineAMD** [comment]
Seeing some failures on CI that I can reproduce locally - I'm guessing it's a memory leak somewhere, maybe I forgot to update a handle somewhere or something, will look into this. Seeing the error on develop too.

**Pull request merged**